src/ensembl/src/content/app/entity-viewer/gene-view/components/protein-domain-image/ProteinDomainImage.tsx

```diff
@@ -102,26 +103,31 @@ const ProteinDomainImage = (props: ProteinDomainImageProps) => {
     };
   }, [props.transcriptId]);
 
-  return data ? <ProteinDomainImageWithData {...props} protein={data} /> : null;
+  return product?.protein_domains_resources ? (
+    <ProteinDomainImageWithData
+      {...props}
+      proteinDomains={product.protein_domains_resources}
+    />
+  ) : null;
 };
 
 export const ProteinDomainImageWithData = (
   props: ProteinDomainImageWithDataProps
 ) => {
-  const { protein } = props;
+  const { proteinDomains, trackLength } = props;
 
-  if (!protein?.protein_domains_resources) {
-    return null;
-  }
+  const proteinDomainsResources = getDomainsByResourceGroups(proteinDomains);
 
-  const proteinDomainsResources = getDomainsByResourceGroups(
-    protein.protein_domains_resources
-  );
+  // If we consider the image's starting point as A and end point as B the scale can be sketched as:
+  // A--  longest protein's length  --B
+  // A--      width in pixels       --B
```

Comment:
same comment as in the protein component

Reply:
✔️